os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -332,31 +331,26 @@ void binary_manager_recovery(int bin_idx)
 	}
 
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
-	/* If a fault happens in common library, we need to reload the library and all user binaries */
+	/* If a fault happens in common library or user binaries, we need to reload the library and all user binaries */
 	int bidx;
-	if (bin_idx == BM_CMNLIB_IDX) {
-		int bin_count = binary_manager_get_ucount();
-		for (bidx = 1; bidx <= bin_count; bidx++) {
-			/* Exclude its all children from scheduling if the binary is registered with the binary manager */
-			ret = binary_manager_deactivate_binary(bidx);
-			if (ret != OK) {
-				bmlldbg("Failure during recovery excluding binary pid = %d\n", bidx);
-				goto reboot_board;
-			}
-
-			BIN_STATE(bidx) = BINARY_FAULT;
-		}
-	} else
-#endif
-	{
+	int bin_count = binary_manager_get_ucount();
+	for (bidx = 1; bidx <= bin_count; bidx++) {
 		/* Exclude its all children from scheduling if the binary is registered with the binary manager */
-		ret = binary_manager_deactivate_binary(bin_idx);
+		ret = binary_manager_deactivate_binary(bidx);
 		if (ret != OK) {
-			bmlldbg("Failed to deactivate binary\n");
+			bmlldbg("Failed to deactivate binary, bin idx %d\n", bidx);
```

Comment:
You used some failure logs with `bmdbg`, but here you used `bmlldbg`.
Is this intended?

Reply:
Yes, it is executed by fault handler.

Reply:
Yes, it is called fault handler